### ISSUE TITLE
[New rule] avoid-missing-enum-constant-in-map

### ISSUE THREAD

**Issue by dened** [dart-code-checker/dart-code-metrics#562]
Because dart doesn't have a enum with value we often use extension with different maps which allow to extend enums.

It will be cool to have a rules that checks that we define all enum values in maps

[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

```dart
enum CountyCode {
  russia,
  kazachstan, // when we add a new value it's a potential bag if we forget to add the value in maps(_code and _title)
  another,
}

extension CountyTypeX on CountyCode {
  // ignore: unused_field
  static const _code = <CountyCode, String>{
    CountyCode.russia: 'RUS',
    CountyCode.another: 'XXX',
  };

  // ignore: unused_field
  static const _title = <CountyCode, String>{
    CountyCode.russia: 'Россия',
    CountyCode.another: 'Другое',
  };

  static CountyCode? fromCode(String? code) => enumDecodeNullable(_code, code);

  String get title => _title[this]!;

  String get code => _code[this]!;
}


```

**Comment by incendial** [comment]
Available in 4.7.0 release 🚀
